etna/records/views/records.py

```diff
@@ -19,25 +20,30 @@ def record_disambiguation_view(request, reference_number):
 
     https://discovery.nationalarchives.gov.uk/browse/r/h/C4122893
     """
-    pages = Record.search.filter(web_reference=reference_number)
+    per_page = 20
+    page_number = int(request.GET.get("page", 1))
+    offset = (page_number - 1) * per_page
 
-    if len(pages) == 0:
+    count, records = Record.api.search(
+        web_reference=reference_number, offset=offset, size=per_page
+    )
+
+    if len(records) == 0:
         raise Http404
```

Comment:
Can the check for Http404 be handled in APIManager like fetch

Reply:
Hi @TNA-Allan. Thanks for taking a look at this PR!

In this instance, we're actually raising a `Http404` exception if Kong response contains no results for the fetch by `webReference`.

In other words, if there are no records belonging to the `webReference` respond with a 404.

Reply:
How is the Http404 rendered to the user?

Reply:
So I believe 404 file was not found , but here we are saying the record is not found so the actual status code is 200 , but raising 404 file not found. I believe we should have an exception class with, maybe RecordNotFoundException